[WIP] 474 feature py3

duk3luk3 opened the pull request:
PR opened mainly to coax travis

Initial PR:

- [ ] PR branch should be named `issuenum`-`fix`/`feature`/`cleanup`-`description`
- [ ] Code is split into logical commits
- [ ] Code has tests
- [ ] Final commit includes "Fixes #issue" in commit message

When all builds pass and a maintainer is happy with your PR, the "ready" label will be applied. Please complete these tasks then:

- [ ] Rebase onto develop
- [ ] Add changelog entry
- [ ] Remove this entire template section

duk3luk3 committed:
Adjust readme for python3 and qt5

duk3luk3 commented:
@muellni @Brutus5000 @Sheeo travis build works now (will migrate the https://github.com/duk3luk3/faf-pyclient-builddeps/ repo when I merge this), but I need help with the windows build - I have no idea why it doesn't work, I just copied @muellni's improved python2 appveyor build. Maybe it needs a different build environment.

muellni commented:
I'll look into the appveyor build.

Commit:
adapt setup.py for py3

muellni commented:
Appveyor builds an installer now, but it's not running yet. Let me know if you need more help.

duk3luk3 commented:
I get this in a pretty much pristine windows 7 virtualbox (no fa installed):

```
---------------------------
FAForever.exe - Entry Point Not Found
---------------------------
The procedure entry point ucrtbase.terminate could not be located in the dynamic link library api-ms-win-crt-runtime-l1-1-0.dll. 
---------------------------
OK   
---------------------------
```

duk3luk3 commented:
On my windows 7 desktop, it just doesn't start up. This is all that's written to the log:

```
2016-11-27 19:18:58,818 INFO     root                           FAF version: 0.11.60-128-g0015698  + 1063 Environment: production
```

Commit:
initial port from pyqt4 to pyqt5

Commit:
662 make tests run

Wesmania committed:
Pass code through 2to3

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Manual py3 fixes

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Unversion cx_freeze in requirements for py3

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Try to fixup cx_freeze

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Remove the SSL disable workaround

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Commit:
move ClientWindows ProgressDialog to cleanup(), since it's only used there

Commit:
remove enum34 requirement

Commit:
hide window titlebar

GrotheFAF commented on src/client/_clientwindow.py:
```diff
@@ -647,66 +637,69 @@ def cleanup(self):
-        self.progress.setCancelButton(None)
-        self.progress.show()
+        progress = QtWidgets.QProgressDialog()
+        progress.setMinimum(0)
```
why is setMinimun and setMaximum there twice?

Wesmania replied:
Yeah, once should be enough. We can make a commit to fix that with  'git commit --fixup <original commit sha>', and we'll meld it in once we start rebasing the branch.

Commit:
make console logging a setting (which defaults to true for development environment)

Wesmania committed:
Convert news widget to QtWebEngine

Also delete old ui file that's no longer used.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Convert unitDB tab to QtWebEngine

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Convert leaderboards tab to QtWebEngine

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Convert tournaments tab to QtWebEngine

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Convert mapvault to QtWebEngine

This involves adding a file on the server that isn't in any CVS
(maps_qt5.php). We'll eventually replace the map vault anyway.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Remove last unused bit of QtWebKit from client window

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Copy pywin32 dlls

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Workaround cx_freeze >5.0.0 not including idna

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Copy over QtWebEngine files

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Manual qt5 fixups

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Fix travis build for py3 / qt5

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Fix appveyor build for py3 / qt5

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Add changelog entry for #509

Fixes #412, fixes #474, fixes #479.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Remove unused widget variable

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>
